share/spice/dota2/dota2.js

```diff
@@ -0,0 +1,190 @@
+                    }
+                );
+            }
+        });
+        return abilities;
+    }
+
+    function getHeroPrimaryAttribute(hero) {
+        switch (hero.pa) {
+            case 'int':
+                return 'Intelligence';
+            case 'str':
+                return 'Strength';
+            case 'agi':
+                return 'Agility';
+        }
+    }
+
+    function getHeroDisplayAssets() {
+        return {
+            stats: {
+                intelligence: {
+                    image: DDG.getImageProxyURL('http://cdn.dota2.com/apps/dota2/images/heropedia/overviewicon_int.png'),
+                    title: 'Intelligence'
+                },
+                agility: {
+                    image: DDG.getImageProxyURL('http://cdn.dota2.com/apps/dota2/images/heropedia/overviewicon_agi.png'),
+                    title: 'Agility'
+                },
+                strength: {
+                    image: DDG.getImageProxyURL('http://cdn.dota2.com/apps/dota2/images/heropedia/overviewicon_str.png'),
+                    title: 'Strength'
+                },
+                attack: {
+                    image: DDG.getImageProxyURL('http://cdn.dota2.com/apps/dota2/images/heropedia/overviewicon_attack.png'),
+                    title: 'Attack'
+                },
+                speed: {
+                    image: DDG.getImageProxyURL('http://cdn.dota2.com/apps/dota2/images/heropedia/overviewicon_speed.png'),
+                    title: 'Speed'
+                },
+                defense: {
+                    image: DDG.getImageProxyURL('http://cdn.dota2.com/apps/dota2/images/heropedia/overviewicon_defense.png'),
+                    title: 'Defense'
+                },
+            }
+        }
+    }
+
+    function createHeroReturnObject(hero, result) {
+        return {
+            isHero: true,
+            name: hero.dname,
+            image: DDG.getImageProxyURL('http://cdn.dota2.com/apps/dota2/images/heroes/' + hero.heroKey + '_vert.jpg'),
+            attackType: hero.dac,
+            primaryAttribute: getHeroPrimaryAttribute(hero),
+            int: hero.attribs.int.b + ' + ' + hero.attribs.int.g,
+            agi: hero.attribs.agi.b + ' + ' + hero.attribs.agi.g,
+            str: hero.attribs.str.b + ' + ' + hero.attribs.str.g,
+            attack: hero.attribs.dmg.min + ' - ' + hero.attribs.dmg.max,
+            speed: hero.attribs.ms,
+            defense: hero.attribs.armor,
+            roles: getHeroRoles(hero),
+            abilities: getHeroAbilities(hero, result),
+            assets: getHeroDisplayAssets()
+        };
+    }
+
+    env.ddg_spice_dota2 = function(api_result){
+
+        if (api_result.error) {
+            return Spice.failed('dota2');
+        }
+
+        var searchTarget = DDG.get_query().replace("dota2", "").replace("dota 2", "").trim().toLowerCase();
```

Comment:
I think we can simplify this a bit: `.replace("dota2", "").replace("dota 2", "")` -> `.replace(/dota\s?2/, "")`